platform/api/oci/instance.go

```diff
@@ -30,7 +31,7 @@ func (a *API) CreateInstance(name, userdata, sshKey string) (*Machine, error) {
 		return nil, err
 	}
 
-	subnet, err := a.getSubnetOnVCN(vcn.ID)
+	subnet, err := a.getSubnetOnVCN(*vcn.Id)
```

Comment:
check for nil for vcn.Id, or if nil is OK in this context, drop a comment saying so

Reply:
Whoops, forgot to add these. Forgot that mandatory fields in JSON do not mean that the fields will not be nil during unmarshaling.